ee/server/service/software_installers.go

```diff
@@ -80,12 +86,26 @@ func (svc *Service) UploadSoftwareInstaller(ctx context.Context, payload *fleet.
 
 	// TODO: QA what breaks when you have a software title with no versions?
 
+	var teamName string
+	if payload.TeamID != nil {
+		t, err := svc.ds.Team(ctx, *payload.TeamID)
+		if err != nil {
+			return err
+		}
+		teamName = t.Name
+	}
+
+	// Create activity
+	if err := svc.ds.NewActivity(ctx, vc.User, fleet.ActivityTypeAddedSoftware{SoftwareTitle: title, SoftwarePackage: payload.Filename, TeamName: teamName}); err != nil {
+		return nil
+	}
```

Comment:
sorry if this wasn't specified in the Figma. For all the activities we generally include the team id as well